# **The Wheat Seeds Dataset**

**Overview**

The Wheat Seeds Dataset involves the prediction of species given measurements of seeds from different varieties of wheat.

It is a binary (2-class) classification problem. 
The number of observations for each class is balanced. 
There are 210 observations with 7 input variables and 1 output variable. 
The variable names are as follows:

- Area.
- Perimeter.
- Compactness
- Length of kernel.
- Width of kernel.
- Asymmetry coefficient.
- Length of kernel groove.
- Class (1, 2, 3).

**Notes: Please make a copy before start working**

**Import tools set**

In [0]:
# Your code here
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

sns.set_style("whitegrid")

**Read 08_seeds.csv as a dataframe call se**

In [0]:
# github link: 'https://raw.githubusercontent.com/dhuy237/Wheat-Seeds/master/08_seeds.csv'
# google drive: 'https://drive.google.com/file/d/1113DaNnvaT1GKkC-26KufoldZ2J9cn78/view'
url = 'https://raw.githubusercontent.com/dhuy237/Wheat-Seeds/master/08_seeds.csv'
se = pd.read_csv(url)

## Checking the data

**Check the head of the DataFrame**

In [10]:
se.head()

Area  Perimeter  Compactness  ...  Asymmetry.Coeff  Kernel.Groove  Type
0  15.26      14.84       0.8710  ...            2.221          5.220     1
1  14.88      14.57       0.8811  ...            1.018          4.956     1
2  14.29      14.09       0.9050  ...            2.699          4.825     1
3  13.84      13.94       0.8955  ...            2.259          4.805     1
4  16.14      14.99       0.9034  ...            1.355          5.175     1

[5 rows x 8 columns]

**Let's see our data's information**

In [11]:
se.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 199 entries, 0 to 198
Data columns (total 8 columns):
Area               199 non-null float64
Perimeter          199 non-null float64
Compactness        199 non-null float64
Kernel.Length      199 non-null float64
Kernel.Width       199 non-null float64
Asymmetry.Coeff    199 non-null float64
Kernel.Groove      199 non-null float64
Type               199 non-null int64
dtypes: float64(7), int64(1)
memory usage: 12.5 KB


**We will have a look at my data features' statistics**

In [13]:
se.describe()

Area   Perimeter  ...  Kernel.Groove        Type
count  199.000000  199.000000  ...     199.000000  199.000000
mean    14.918744   14.595829  ...       5.420653    1.994975
std      2.919976    1.310445  ...       0.492718    0.813382
min     10.590000   12.410000  ...       4.519000    1.000000
25%     12.330000   13.470000  ...       5.046000    1.000000
50%     14.430000   14.370000  ...       5.228000    2.000000
75%     17.455000   15.805000  ...       5.879000    3.000000
max     21.180000   17.250000  ...       6.550000    3.000000

[8 rows x 8 columns]

## Cleaning the data

**Check for NaN values**

In [14]:
se.isnull().sum()

Area               0
Perimeter          0
Compactness        0
Kernel.Length      0
Kernel.Width       0
Asymmetry.Coeff    0
Kernel.Groove      0
Type               0
dtype: int64

## EDA

The last column is reserved for the Kernel type. This particular dataset has 199 entries. Some of these variables are explicitly dependent. 

For example, compactness: C = 4piArea/(Perimeter)^2 has a linear proportional relationship with area, and also a square proportionality with kernel width.